extending-the-rest-api/adding-rest-api-support-for-custom-content-types.md

```diff
@@ -128,55 +128,52 @@ function my_book_taxonomy() {
 ```
 
 ## Adding REST API Support To Existing Content Types
-When a custom post type or custom taxonomy has been added by code that you do not control, for example a theme or plugin you are using, you may need to add REST API support after it has already been registered. The arguments are the same as in the previous examples, but need to be added to the global `$wp_post_types` and `$wp_taxonomies` arrays.
 
-Here is an example of adding REST API support to an existing custom post type:
+If you need to add REST API support for a custom post type or custom taxonomy you do not control, for example a theme or plugin you are using, you can use the `register_post_type_args` filter hook that exists since WordPress version 4.6.0.
 
 ```php
 /**
  * Add REST API support to an already registered post type.
  */
-add_action( 'init', 'my_custom_post_type_rest_support', 25 );
-function my_custom_post_type_rest_support() {
-  global $wp_post_types;
-
-  //be sure to set this to the name of your post type!
-  $post_type_name = 'planet';
-  if( isset( $wp_post_types[ $post_type_name ] ) ) {
-    $wp_post_types[$post_type_name]->show_in_rest = true;
-    // Optionally customize the rest_base or controller class
-    $wp_post_types[$post_type_name]->rest_base = $post_type_name;
-    $wp_post_types[$post_type_name]->rest_controller_class = 'WP_REST_Posts_Controller';
-  }
+add_filter( 'register_post_type_args', 'my_post_type_args', 10, 2 );
+
+function my_post_type_args( $args, $post_type ) {
+
+	if ( 'book' === $post_type ) {
+		$args['show_in_rest']          = true;
+        
+        // Optionally customize the rest_base or rest_controller_class
```

Comment:
Your indentation is off here, and the line above has extra whitespace. We should be using tabs throughout :)